src/main.rs

```diff
@@ -116,7 +119,8 @@ impl<'a> From<&'a str> for Host {
     }
 }
 
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize)]
+#[serde(from = "&str")]
```

Comment:
why from `&str` and not `String` ?

Reply:
Since `Host` already implements `From<&str>` and `Target` implements `From<Host>`, I added the transitive `From<&str> for Target`, which is then used by `serde` to deserialize the keys of the map [here](https://github.com/mntns/cross/blob/ea276c1f653b210ebee76591aac3a00b40966874/src/cross_toml.rs#L40). I think there is no big difference between using `String` or a string slice for the deserialization.